lib/frodo/query.rb

```diff
@@ -29,13 +29,8 @@ def [](property)
     # Build a query to find an entity with the supplied key value.
     # @param key [to_s] primary key to lookup
     # @return the path and querystring [String]
-    def find(key)
-      entity = @entity_set.new_entity
-      key_property = entity.get_property(entity.primary_key)
-      key_property.value = key
-
-      pathname = "#{entity_set.name}(#{key_property.url_value})"
-
+    def find(key, entity_set_name)
+      pathname = "#{entity_set_name}(#{key})"
```

Comment:
Does `entity_set.name` not work? Could eliminate the parameter then

Reply:
you right we can just use entity_set.name, because it will be initialized on